performance_history_analysis.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+from timeit import default_timer as timer
+import git
+import typer
+from rich import print
+
+
+def analyse_performance_history(
+    rev: str, limit_commits: int, sample_amount: int, print_all_results: bool
+):
+    limit = limit_commits
+    repo = git.Repo()
+
+    for commit in repo.iter_commits(rev=rev):
```

Comment:
Instead of using `limit` you can pass `max_count` kwarg to `iter_commits`

Reply:
Thanks, nice tip! Changed.